roles/elasticsearch/tasks/main.yml

```diff
@@ -1,16 +1,199 @@
 ---
+- name: check if certs exist
+  stat: 'path=/etc/letsencrypt/live/{{ hostname }}/privkey.pem'
+  register: certfile
+
+- name: fail if certs not present
+  fail: 'msg="/etc/letsencrypt/live/{{ hostname }}/privkey.pem not found"'
+  when: not certfile.stat.exists
+
+- name: set vars
+  ansible.builtin.set_fact:
+    elastic_running: false
+    cluster_running: false
+
+- name: check if elasticsearch already running
+  ansible.builtin.uri:
+    url: 'https://{{ hostname }}:{{ http_port }}/'
+    user: '{{ admin_user }}'
+    password: '{{ admin_password }}'
+    validate_certs: false
+    return_content: true
+  register: elastic_response
+  ignore_errors: true
+
+- name: set a var if ES is already running
+  when: "elastic_response.content is defined and 'cluster_name' in elastic_response.content"
+  ansible.builtin.set_fact:
+    elastic_running: true
+
 - name: install elasticsearch rpm gpg key
+  when: elastic_running is false
   rpm_key: state=present key=https://artifacts.elastic.co/GPG-KEY-elasticsearch
 
 - name: install elasticsearch 8.x repository
   copy: src=elasticsearch-8.repo dest=/etc/yum.repos.d/elasticsearch.repo owner=root group=root mode=0644
-  when: elasticsearch_version == 8
+  when: elasticsearch_version == 8 and elastic_running is false
 
 - name: install elasticsearch and dependencies
+  when: elastic_running is false
   yum: name={{item}} state=latest
   with_items:
     - elasticsearch
   notify: update aide
 
+- name: open http port
+  ansible.builtin.shell: 'firewall-cmd --zone=public --add-port={{ http_port }}/tcp'
+
+- name: open transport port
+  ansible.builtin.shell: 'firewall-cmd --zone=public --add-port={{ transport_port }}/tcp'
+
+- name: create old certs dir
+  ansible.builtin.file:
+    path: /etc/elasticsearch/certs/old
+    state: directory
+    mode: '0750'
+
+- name: old certs dir ownership
+  ansible.builtin.file:
+    path: /etc/elasticsearch/certs/old
+    state: directory
+    owner: elasticsearch
+    group: elasticsearch
+
+- name: copy certs script
+  ansible.builtin.template:
+    src: es-certs.sh
+    dest: /usr/local/sbin/es-certs.sh
+    owner: root
+    group: root
+    mode: '0755'
+
+- name: run certs script
+  ansible.builtin.shell: /usr/local/sbin/es-certs.sh
+
+- name: copy elasticsearch.yml
+  ansible.builtin.template:
+    src: elasticsearch.yml
+    dest: /etc/elasticsearch/elasticsearch.yml
+    owner: elasticsearch
+    group: elasticsearch
+
+- name: copy roles.yml
+  ansible.builtin.template:
+    src: roles.yml
+    dest: /etc/elasticsearch/roles.yml
+    owner: elasticsearch
+    group: elasticsearch
+
+- name: uncomment initial master nodes
+  when: elastic_running is false
+  ansible.builtin.replace:
+    path: /etc/elasticsearch/elasticsearch.yml
+    backup: true
+    regexp: '^ *#cluster\.initial_master_nodes'
+    replace: '  cluster.initial_master_nodes'
```

Comment:
why do this here instead of just having the template elasticsearch.yml with the values?

Reply:
Because we only do this when we're bootstrapping a new cluster, and it will cause issues if we later sync an updated yml file to a running node with this uncommented